services/harmony/app/middleware/shapefile-converter.ts

```diff
@@ -81,6 +84,103 @@ const contentTypesToConverters = {
+ */
+function normalizeLongitude(lon: number): number {
+  while (lon > 180) lon -= 360;
+  while (lon < -180) lon += 360;
+  return lon;
+}
+
+/**
+ *
+ * @param geojson - normalize all the longitudes in the file to [-180,180]
+ * @returns
+ */
+export function normalizeGeoJsonCoords(geojson: any): any {
+  function normalizeCoordinates(coordinates: any): any {
+    if (Array.isArray(coordinates[0])) {
+      return coordinates.map(normalizeCoordinates);
+    } else {
+      return [normalizeLongitude(coordinates[0]), coordinates[1]];
+    }
+  }
+
+  function normalizeGeometry(geometry: any): any {
+    if (geometry.type === 'Point') {
+      geometry.coordinates = normalizeCoordinates(geometry.coordinates);
+    } else if (geometry.type === 'MultiPoint' || geometry.type === 'LineString') {
+      geometry.coordinates = geometry.coordinates.map(normalizeCoordinates);
+    } else if (geometry.type === 'Polygon' || geometry.type === 'MultiLineString') {
+      geometry.coordinates = geometry.coordinates.map(ring => ring.map(normalizeCoordinates));
+    } else if (geometry.type === 'MultiPolygon') {
+      geometry.coordinates = geometry.coordinates.map(polygon => polygon.map(ring => ring.map(normalizeCoordinates)));
+    } else if (geometry.type === 'GeometryCollection') {
+      geometry.geometries = geometry.geometries.map(normalizeGeometry);
+    }
+    return geometry;
+  }
+
+  function normalizeFeature(feature: any): any {
+    if (feature.type === 'Feature') {
+      feature.geometry = normalizeGeometry(feature.geometry);
+    } else if (feature.type === 'FeatureCollection') {
+      feature.features = feature.features.map(normalizeFeature);
+    }
+    return feature;
+  }
+
+  return normalizeFeature(geojson);
+}
+
+/**
+ * Change longitudes of a geojson file to be in the [-180, 180] range and split at antimeridian
+ * if needed
+ * @param geoJson - An object representing the json for a geojson file
+ * @returns An object with the normalized geojson
+ */
+export function normalizeGeoJson(geoJson: object): object {
+  let newGeoJson = normalizeGeoJsonCoords(geoJson);
+
+  // eslint-disable-next-line @typescript-eslint/dot-notation
+  for (const index in newGeoJson['features']) {
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    const feature = newGeoJson['features'][index];
+    const normalizedGeoJson = splitGeoJson(feature);
+    // eslint-disable-next-line @typescript-eslint/dot-notation
+    newGeoJson['features'][index] = normalizedGeoJson;
+  }
+
+  // force ccw winding
+  newGeoJson = rewind(newGeoJson, false);
+  return newGeoJson;
+}
+
+/**
+ * Handle any weird cases like splitting geometry that crosses the antimeridian
+ * @param url - the url of the geojson file
+ * @param isLocal - whether the url is a downloaded file (true) or needs to be downloaded (false)
+ */
+async function normalizeGeoJsonFile(url: string, isLocal: boolean, logger: Logger): Promise<string> {
+  logger.debug('Normalizing geojson');
```

Comment:
This message doesn't look to useful unless we're going to log some info about the geometry.

Reply:
Removed